test/parallel/test-buffer-swap.js

```diff
@@ -3,13 +3,26 @@
 require('../common');
 const assert = require('assert');
 
-const buf = Buffer.from([0x1, 0x2, 0x3, 0x4]);
+// Test buffers small enough to use the JS implementation
+const buf = Buffer.from([0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09,
+                         0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f, 0x10]);
 
 assert.strictEqual(buf, buf.swap16());
-assert.deepStrictEqual(buf, Buffer.from([0x2, 0x1, 0x4, 0x3]));
+assert.deepStrictEqual(buf, Buffer.from([0x02, 0x01, 0x04, 0x03, 0x06, 0x05, 0x08,
+                                         0x07, 0x0a, 0x9, 0x0c, 0x0b, 0x0e, 0x0d,
+                                         0x10, 0x0f]));
+buf.swap16(); // restore
 
 assert.strictEqual(buf, buf.swap32());
-assert.deepStrictEqual(buf, Buffer.from([0x3, 0x4, 0x1, 0x2]));
+assert.deepStrictEqual(buf, Buffer.from([0x04, 0x03, 0x02, 0x01, 0x08, 0x07, 0x06,
+                                         0x05, 0x0c, 0x0b, 0x0a, 0x09, 0x10, 0x0f,
+                                         0x0e, 0x0d]));
+buf.swap32(); // restore
+
+assert.strictEqual(buf, buf.swap64());
+assert.deepStrictEqual(buf, Buffer.from([0x08, 0x07, 0x06, 0x05, 0x04, 0x03, 0x02,
+                                         0x01, 0x10, 0x0f, 0x0e, 0x0d, 0x0c, 0x0b,
+                                         0x0a, 0x09]));
```

Comment:
Add tests that use a slice of a buffer with offset of 1. This way alignment will be off, and can make sure the code runs properly regardless.